### PR TITLE
Feature: handle expired token

### LINKED ISSUE #478: Redo login if token is expired
## Description

Currently, if the token is expired, Leto is blocked at the splash screen.
The goal of this task is to redo the login process if the token is expired.

So what we need to do is, to delete the old token, redo the login process and store the new token.

### PR DESCRIPTION
**Pull request by aboubakar14**
resolve #478

### CODE REVIEW

File: src/pages/SplashPage.vue [hunk 1]
```diff
@@ -78,12 +78,19 @@ async function initUser() {
...
-          type: 'negative',
-          message: t('errors.authentication.fetchingData'),
-          html: true,
-        });
+      .catch((error) => {
+        const errorData = error.response?.data;
+        // Code 209 == Token Expired
+        if (errorData?.code === 209) {
+          removeUserSessionToken();
+          router.push({ name: 'Splash' });
```
Comment (Zorin95670) [inline src/pages/SplashPage.vue:86]: you should push with the same parameters as you received.

**Review by Zorin95670 — DISMISSED** [review]

File: src/pages/SplashPage.vue [hunk 2]
```diff
@@ -78,12 +78,19 @@ async function initUser() {
...
     await initUserInformation(getUserSessionToken())
-      .catch(() => {
-        Notify.create({
-          type: 'negative',
-          message: t('errors.authentication.fetchingData'),
-          html: true,
-        });
+      .catch((error) => {
+        const errorData = error.response?.data;
+        // Code 209 == Token Expired
```
Comment (k-y0u) [inline src/pages/SplashPage.vue:83]: jump before

File: tests/unit/pages/SplashPage.spec.js [hunk 3]
```diff
@@ -161,5 +162,24 @@ describe('Test component: SplashPage', () => {
...
+            data: { code: 209 },
+          },
+        },
+      ));
+      setActivePinia(createPinia());
+      Notify.create = jest.fn();
+
+      await wrapper.vm.initUser();
+
+      expect(getUserSessionToken).toHaveBeenCalledTimes(4);
```
Comment (k-y0u) [inline tests/unit/pages/SplashPage.spec.js:180]: Why four times ?

**Review by k-y0u — CHANGES_REQUESTED** [review]

File: src/pages/SplashPage.vue (hunk 2 above)
Comment (aboubakar14) [reply src/pages/SplashPage.vue:84]: Done

File: tests/unit/pages/SplashPage.spec.js (hunk 3 above)
Comment (aboubakar14) [reply tests/unit/pages/SplashPage.spec.js:180]: Because if everything went well the getUserSessionToken is called 4 times

**Review by justinehell — APPROVED** [review]

**Review by Zorin95670 — APPROVED** [review]

**Review by k-y0u — APPROVED** [review]

**Pull request merged**